lstchain/calib/camera/drs4.py

```diff
@@ -42,6 +42,7 @@ def __init__(self, tel_id, n_module, **kwargs):
         self.meanped = np.zeros((n_gain, self.n_pixels, size4drs))
         self.numped = np.zeros((n_gain, self.n_pixels, size4drs))
         self.first_cap_array = np.zeros((self.n_module, n_gain, n_channel))
+        self.failing_pixels_list = None
```

Comment:
will this be a list of an array?
what about initializing it as an empty list  `self.failing_pixels_list = []`?

Reply:
If `failing_pixels_list` is no empty fits file will contain numpy ndarray (`fits_file[2].data`), otherwise value will be empty  (`NoneType`).

Reply:
then what about initializing it as an empty array, and calling it `failing_pixels_array`?

Reply:
> then what about initializing it as an empty array, and calling it `failing_pixels_array`?

return None.

Reply:
no, what I mean is that since it will be an array later, it may not be the best idea to initialize it as a `None`, but as a `np.empty` array.
Also, I'm not sure if a variable length array is the best here, or maybe better to have a boolean array:

`failing_pixels_array  = numpy.full((self.n_pixels), False)` 

and set it `True` only for the failing pixels

Reply:
> fits_file[2].data

Please use extnames to identify tables instead of indices.

Reply:
> no, what I mean is that since it will be an array later, it may not be the best idea to initialize it as a `None`, but as a `np.empty` array.
> Also, I'm not sure if a variable length array is the best here, or maybe better to have a boolean array:
> 
> `failing_pixels_array = numpy.full((self.n_pixels), False)`
> 
> and set it `True` only for the failing pixels

Done.

Reply:
> > fits_file[2].data
> 
> Please use extnames to identify tables instead of indices.

Done